src/plugins/ucx/ucx_backend.cpp

```diff
@@ -359,34 +360,50 @@ void nixlUcxEngine::progressFunc()
     }
     pthrActiveCV.notify_one();
 
+    // Set POLLIN event on all worker fds so that the main loop would process them all on first iteration
+    for (size_t wid = 0; wid < pollFds.size() - 1; wid++)
+        pollFds[wid].revents = POLLIN;
+
+    bool pthrStop = false;
     while (!pthrStop) {
-        int i;
-        for(i = 0; i < noSyncIters; i++) {
-            for (auto &uw: uws)
-                uw->progress();
+        for (size_t wid = 0; wid < pollFds.size() - 1; wid++) {
```

Comment:
is it from #297?

Reply:
It seems that every time I force-push the branch that this PR #297 is based on I also need to rebase and push this one, otherwise Github treats outdated base branch commit as belonging to this PR. Fixed it now, thanks.